views/js/previewer/provider/item.js

```diff
@@ -0,0 +1,388 @@
+                .on('submititem', function () {
+                    var itemRunner = this.itemRunner;
+                    var itemState = itemRunner.getState();
+                    var itemResponses = itemRunner.getResponses();
+
+                    this.trigger('disabletools disablenav');
+                    this.trigger('submitresponse', itemResponses, itemState);
+
+                    return this.getProxy()
+                        .submitItem(dataHolder.get('itemIdentifier'), itemState, itemResponses)
+                        .then(function submitSuccess(response) {
+                            self.trigger('scoreitem', response);
+                            self.trigger('enabletools enablenav resumeitem');
+                        })
+                        .catch(function submitError(err) {
+                            self.trigger('enabletools enablenav');
+
+                            //some server errors are valid, so we don't fail (prevent empty responses)
+                            if (err.code === 200) {
+                                self.trigger('alert.submitError',
+                                    err.message || __('An error occurred during results submission. Please retry.'),
+                                    function () {
+                                        self.trigger('resumeitem');
+                                    }
+                                );
+                            }
+                        });
+                })
+                .on('ready', function () {
+                    var itemIdentifier = dataHolder.get('itemIdentifier');
+                    var itemData = dataHolder.get('itemData');
+
+                    if (itemIdentifier) {
+                        if (itemData) {
+                            self.renderItem(itemIdentifier, itemData);
+                        } else {
+                            self.loadItem(itemIdentifier);
+                        }
+                    }
+                })
+                .on('loaditem', function (itemRef, itemData) {
+                    dataHolder.set('itemIdentifier', itemRef);
+                    dataHolder.set('itemData', itemData);
+                })
+                .on('renderitem', function () {
+                    this.trigger('enabletools enablenav');
+                })
+                .on('resumeitem', function () {
+                    this.trigger('enableitem enablenav');
+                })
+                .on('disableitem', function () {
+                    this.trigger('disabletools');
+                })
+                .on('enableitem', function () {
+                    this.trigger('enabletools');
+                })
+                .on('error', function () {
+                    this.trigger('disabletools enablenav');
+                })
+                .on('finish leave', function () {
+                    this.trigger('disablenav disabletools');
+                    this.trigger('endsession');
```

Comment:
The 2 events `endsession` and `beforeunloaditem` are artifacts kept only for compatibility (as far as I remember in the test runner provider), not sure the item previewer should support them